proposals/0006-resource-attributes.md

```diff
@@ -0,0 +1,82 @@
+| Texture2DArray                  | SRV                  | -            | 2                       | -             | -           | -                 | yes            |
+| RWTexture2DArray                | UAV                  | -            | 2                       | -             | -           | -                 | yes            |
+| RasterizerOrderedTexture2DArray | UAV                  | yes          | 2                       | -             | -           | -                 | yes            |
+| Texture3D                       | SRV                  | -            | 3                       | -             | -           | -                 | -              |
+| RWTexture3D                     | UAV                  | -            | 3                       | -             | -           | -                 | -              |
+| RasterizerOrderedTexture3D      | UAV                  | yes          | 3                       | -             | -           | -                 | -              |
+| TextureCUBE                     | SRV                  | -            | 3                       | yes           | -           | -                 | -              |
+| TextureCUBEArray                | SRV                  | -            | 3                       | yes           | -           | -                 | yes            |
+| Texture2DMS                     | SRV                  | -            | 2                       | -             | yes         | -                 | -              |
+| Texture2DMSArray                | SRV                  | -            | 2                       | -             | yes         | -                 | yes            |
+| RWTexture2DMS                   | UAV                  | -            | 2                       | -             | yes         | -                 | -              |
+| RWTexture2DMSArray              | UAV                  | -            | 2                       | -             | yes         | -                 | yes            |
+| FeedbackTexture2D               | SRV                  | -            | 2                       | -             | -           | yes               | -              |
+| FeedbackTexture2DArray          | SRV                  | -            | 2                       | -             | -           | yes               | yes            |
+| TypedBuffer                     | SRV                  | -            | -                       | -             | -           | -                 | -              |
+| RawBuffer                       | SRV                  | -            | -                       | -             | -           | -                 | -              |
+| StructuredBuffer                | SRV                  | -            | -                       | -             | -           | -                 | -              |

```

Comment:
The three buffer types are indistinguishable according to these attributes, and I expect we want to cover the missing `ConstantBuffer` and `TextureBuffer` types here as well.

Here are some ideas for extending the attributes to distinguish buffer types and incorporate `ConstantBuffer`/`TextureBuffer`:

- Add `[[hlsl::row_layout]]` for `ConstantBuffer` and `TextureBuffer` to identify where special cbuffer row layout rules are used with untranslated components from the constant buffer or typed buffer.
- `[[hlsl::row_layout]]` would be combined with `[[hlsl::resource_class(CBV)]]` for ConstantBuffer or `[[hlsl::resource_class(SRV)]]` for TextureBuffer.
- Add `[[hlsl::raw_buffer]]` for `RawBuffer` and `StructuredBuffer`.
- Add `[[hlsl::struct_stride(n)]]` for `StructuredBuffer`, where `n` is the stride for the structured buffer in bytes.